### DISCUSSION TITLE
Is there a way to find if someone is in edit of content?

### DISCUSSION THREAD

**Discussion by tropcicstefan** [OrchardCMS/OrchardCore#7360] [Q&A] [answered]
So my question is, there a feature with which when author is in editor of content item will find out if other author is already editing that content item?

**Comment by jptissot** [comment, marked as answer]
I don't think this feature is currently available. Please feel free to post an issue with this as a feature request.

**Reply by hishamco** [reply]
What you asking is one of the features that planned in Orchard Core Contrib, but it will be great if it's implemented here, I will try to share how this could be done, If I remember there are two strategies that I thought about

**Reply by hishamco** [reply]
1. FIFE

In this strategy the First In the First Edit (FIFE), so whenever a user open the edit page a content will be locked temporary, if any other user try to enter the same page, the system can recognize that the page is edited by someone else, there are prons and cons for this:

- Storage Mechanism

- If the user leave the browser open with no action, perhaps a time out is useful for this case

- The page should be lock if the user start editing the content not open it only

2. Concurrent Access

This strategy may more complex, because by default we allow multiple editing, but we should decide what should be saved at the end

- May required real time communication (SignalR)

- Calculate content diff, something to what I saw in AuditTrail module

**Comment by tropcicstefan** [comment]
Just as a reference, for my project I made custom part that stores contentItemId and username in memory cache in display driver of a part in edit method. Then in shape view I call api(endpoint to handle concurrent edit store) again and check if item is locked by current user just for double checking(in case of slow form opening). After that shape periodically pings store to keep lock.
In case of entering a locked item I hide form and notify user who locked the item. Not much but works for my project for now.
